addons/survey/static/src/js/survey_form.js

```diff
@@ -537,6 +543,74 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
         }
     },
 
+    _showCorrectAnswers: function(correctAnswers, submitPromise, options) {
+        // Display the correct answers
+        Object.keys(correctAnswers).forEach(questionId => this._showQuestionAnswer(correctAnswers, questionId));
+        // Make the form completely readonly
+        const form = document.querySelector('form');
+        form.querySelectorAll('input, textarea, label, td')?.forEach(node => {
+            node.classList.add("pe-none");
+        });
+        // Replace the Submit button by a Next button
+        form.querySelector('button[type="submit"]').classList.add('d-none');
+        const nextPageBtn = form.querySelector('button[id="next_page"]');
+        nextPageBtn.classList.remove('d-none');
+        nextPageBtn.addEventListener('click', () => {
+            this._nextScreen(submitPromise, options);
+        });
+        // Replacing the original onKeyDown listener to block everything except for the
+        // enter or arrow right key down events trigerring the next page display
+        const nextPageKeydownListener = (event) => {
+            if (event.code === 'Enter' || event.code === 'ArrowRight') {
+                // Restore original keydown listener
+                document.removeEventListener('keydown', nextPageKeydownListener);
+                document.addEventListener('keydown', this.documentKeydownListener);
+                this._nextScreen(submitPromise, options);
+            }
+        }
+        document.removeEventListener('keydown', this.documentKeydownListener);
+        document.addEventListener('keydown', nextPageKeydownListener);
+    },
+
+    _showQuestionAnswer: function(correctAnswers, questionId) {
+        const correctAnswer = correctAnswers[questionId];
+        const questionWrapper = document.querySelector(`.js_question-wrapper[id="${questionId}"]`);
+        const answerWrapper = questionWrapper.querySelector('.o_survey_answer_wrapper');
+        const questionType = questionWrapper.querySelector('[data-question-type]').dataset.questionType;
+
+        if (['numerical_box', 'date', 'datetime'].includes(questionType)) {
+            const input = answerWrapper.querySelector('input');
+            let value;
+            if (questionType == 'numerical_box') {
+                value = input.valueAsNumber;
+            } else if (questionType == 'datetime') {
+                const datetime = parseDateTime(input.value);
+                value = datetime ? datetime.setZone("utc").toFormat("MM/dd/yyyy HH:mm:ss", { numberingSystem: "latn" }) : '';
+            } else {
+                value = input.value;
+            }
+            const isCorrect = value == correctAnswer;
+            answerWrapper.classList.add(`bg-${isCorrect ? 'success' : 'danger'}`);
```

Comment:
At it didn't go through before, let's be explicit:

```suggestion
            let isCorrect;
            if (questionType == 'numerical_box') {
                isCorrect = input.valueAsNumber === correctAnswer;
            } else if (questionType == 'datetime') {
                const datetime = parseDateTime(input.value)?.setZone("utc").toFormat("MM/dd/yyyy HH:mm:ss", { numberingSystem: "latn" }) 
                isCorrect = datetime === correctAnswer;
            } else {
                isCorrect = input.value === correctAnswer;
            }
            answerWrapper.classList.add(`bg-${isCorrect ? 'success' : 'danger'}`);
```